### PR TITLE
Replace edit pencil with a button+modal

### LINKED ISSUE #370: Replace edit pencil button with btn-primary-outline Edit button and use modal to edit group and label

### PR DESCRIPTION
**Pull request by SirCmpwn**
Closes #370

### CODE REVIEW

**Comment by eatonphil** [comment]
Looks awesome.

* `s/Save changes/Save/`?
* Just that ^^
* Might need to be concerned about long group names/label names in terms of layout

**Comment by SirCmpwn** [comment]
>s/Save changes/Save/?

I made this change, but fwiw it makes the "yes" button smaller than the "no" button.

File: test/linodes/layouts/LinodeDetailPage.spec.js [hunk 1]
```diff
@@ -334,158 +335,132 @@ describe('linodes/layouts/LinodeDetailPage', () => {
...
-        />);
-      const editor = page.find('.edit-details');
-      expect(editor).to.exist;
-      expect(editor.find('button').length).to.equal(2);
-      expect(editor.find('button.btn-primary').props())
-        .to.have.property('disabled').which.equals(true);
-      expect(editor.find('button.btn-secondary').props())
-        .to.have.property('disabled').which.equals(true);
-    });
+  it('hide modal when "Nevermind" is clicked', () => {
```
Comment (na3d) [inline test/linodes/layouts/LinodeDetailPage.spec.js]: Would it be overkill to make sure that if the input text changes, that after Nevermind is clicked, that nothing changed?

Comment (SirCmpwn) [inline test/linodes/layouts/LinodeDetailPage.spec.js]: Yes

**Comment by na3d** [comment]
👍

**Comment by eatonphil** [comment]
-1 to the border above the buttons, but I'm not seeing where that's coming from

**Comment by SirCmpwn** [comment]
Border was already there on all modals, I just tweaked the style to be less broken. Personally I like the border.

**Comment by eatonphil** [comment]
It's just totally unnecessary. Will file separate issue/proposal for that.

**Pull request merged**